micro-deps/micro-deps-spring-config/src/main/groovy/com/ofg/infrastructure/discovery/DependencyResolutionConfiguration.groovy

```diff
@@ -34,6 +35,9 @@ class DependencyResolutionConfiguration {
     @Autowired(required = false)
     private DependencyPresenceOnStartupVerifier dependencyPresenceOnStartupVerifier
 
+    @Autowired(required = false)
```

Comment:
Why is this dependency optional? And why isn't it declared as a Spring bean? This is how we allow replacing it by end users? Why not use `@Conditional...` or `@Primary`?